frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts

```diff
@@ -161,23 +162,29 @@ export const legacyEntityToNode = (
     }
 
     if (entity.type === 'actions') {
-        return objectCleanWithEmpty({
-            kind: NodeKind.ActionsNode,
-            id: entity.id,
-            ...shared,
-        }) as any
+        return setLatestVersionsOnQuery(
```

Comment:
Just curious, I've been looking through the implementation of `setLatestVersionsOnQuery` (the bit about getting the latest versions for each query is very neat!). Guessing that logic only pulls the latest version for each query and adds it to the passed in query. 
- But if the passed in query is outdated, it would still set the latest version on it? 
- So where ever we do use `setLatestVersionsOnQuery` we need to remember to always keep it the most recent query?

Reply:
Yes, exactly. The assumption we make that the hardcoded queries on the frontend will be adapted by whatever change a migration makes.